Improve the doc-related PR builds

I've broken a `documentation` build with this PR, https://github.com/quarkusio/quarkus/pull/10653#issuecomment-657671432, but the PR build did not detect it, both the PR build and myself were doing only `doc` builds :-).

Proposal: have both `doc` and `documentation` built

@sberyozkin I'm the one that created the doc CI build that should detect such issue so i'm partialy the one to blame ;)

I notice that the build failed with this error:
```
[ERROR] Failed to execute goal org.asciidoctor:asciidoctor-maven-plugin:2.0.0:process-asciidoc (output-pdf) on project quarkus-documentation: Found 92 issue(s) of severity WARN or higher during conversion -> [Help 1]
```

The doc CI build only do a standard doc, but the regular CI seems to generates PDF (it uses `-Ddocumentation-pdf`) and it appears the issue was during PDF generation.

So we need to update the doc CI build to also generates the PDF.

--- a/.github/workflows/doc-build.yml
+++ b/.github/workflows/doc-build.yml
@@ -38,4 +38,7 @@ jobs:
           restore-only: ${{ github.event_name == 'pull_request' }}
       - name: Build
         run: |
-          mvn -e -B clean org.asciidoctor:asciidoctor-maven-plugin:process-asciidoc -pl docs
+          mvn -Dquickly
+      - name: Build Docs
+        run: |
+          mvn -e -B clean org.asciidoctor:asciidoctor-maven-plugin:process-asciidoc -pl docs -Ddocumentation-pdf

